decoder/api_pokemon_scan_v2.go

```diff
@@ -0,0 +1,428 @@
+package decoder
+
+import (
+	"github.com/UnownHash/gohbem"
+	log "github.com/sirupsen/logrus"
+	"golbat/config"
+	"golbat/geo"
+	pb "golbat/grpc"
+	"math"
+	"strconv"
+	"time"
+)
+
+type ApiPokemonScan2 struct {
+	Min        geo.Location          `json:"min"`
+	Max        geo.Location          `json:"max"`
+	Limit      int                   `json:"limit"`
+	DnfFilters []ApiPokemonDnfFilter `json:"filters"`
+}
+
+type ApiPokemonDnfFilter struct {
+	Pokemon []ApiPokemonDnfId     `json:"pokemon"`
+	Iv      *ApiPokemonDnfMinMax8 `json:"iv"`
+	AtkIv   *ApiPokemonDnfMinMax8 `json:"atk_iv"`
+	DefIv   *ApiPokemonDnfMinMax8 `json:"def_iv"`
+	StaIv   *ApiPokemonDnfMinMax8 `json:"sta_iv"`
+	Level   *ApiPokemonDnfMinMax8 `json:"level"`
+	Cp      *ApiPokemonDnfMinMax  `json:"cp"`
+	Gender  *ApiPokemonDnfMinMax8 `json:"gender"`
+	Size    *ApiPokemonDnfMinMax8 `json:"size"`
+	Little  *ApiPokemonDnfMinMax  `json:"pvp_little"`
+	Great   *ApiPokemonDnfMinMax  `json:"pvp_great"`
+	Ultra   *ApiPokemonDnfMinMax  `json:"pvp_ultra"`
+}
+
+type ApiPokemonDnfId struct {
+	Pokemon int16  `json:"id"`
+	Form    *int16 `json:"form"`
+}
+
+type ApiPokemonDnfMinMax struct {
+	Min int16 `json:"min"`
+	Max int16 `json:"max"`
+}
+
+type ApiPokemonDnfMinMax8 struct {
+	Min int8 `json:"min"`
+	Max int8 `json:"max"`
+}
+
+func internalGetPokemonInArea2(retrieveParameters ApiPokemonScan2) []uint64 {
+	type dnfFilterLookup struct {
+		pokemon int16
+		form    int16
+	}
+
+	dnfFilters := make(map[dnfFilterLookup][]ApiPokemonDnfFilter)
+
+	for _, filter := range retrieveParameters.DnfFilters {
+		if len(filter.Pokemon) > 0 {
+			for _, keyString := range filter.Pokemon {
+				var pokemonId int16 = -1
```

Comment:
`keyString.Pokemon`?

Reply:
i don't understand this observation

Reply:
Is this variable supposed to be -1 always? Then why not use a const?

Reply:
Right, I get it now.  Codeblind!